hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupReaderSchemaHandler.java

```diff
@@ -324,4 +404,26 @@ private static Schema.Field getField(Schema schema, String fieldName) {
     }
     return foundFieldOpt.get();
   }
+
+  private Map<String, String> buildCompletionTimeMapping(HoodieTableMetaClient metaClient) {
+    return metaClient.getCommitsTimeline().filterCompletedInstants().getInstants().stream()
```

Comment:
This does not account for instants that are no longer in the active timeline. We'll need to use the `CompletionTimeQueryView` for this